raveandroid/src/main/java/com/flutterwave/raveandroid/RavePayManager.java

```diff
@@ -313,10 +332,22 @@ private AppComponent setUpGraph() {
             baseUrl = LIVE_URL;
         }
 
-        return DaggerAppComponent.builder()
-                .androidModule(new AndroidModule(activity))
-                .networkModule(new NetworkModule(baseUrl))
-                .build();
+        if (activity != null) {
+            return DaggerAppComponent.builder()
+                    .androidModule(new AndroidModule(activity))
+                    .networkModule(new NetworkModule(baseUrl))
+                    .build();
+        } else if (supportFragment != null) {
+            return DaggerAppComponent.builder()
+                    .androidModule(new AndroidModule(supportFragment.getContext()))
+                    .networkModule(new NetworkModule(baseUrl))
+                    .build();
+        } else {
```

Comment:
add another else if statement that checks this

else if `(fragment != null) && (fragment. getActivity() != null)`